RecoLocalMuon/GEMCSCSegment/plugins/GEMCSCSegFit.h

```diff
@@ -0,0 +1,179 @@
+      for(std::map<uint32_t, const CSCLayer*>::const_iterator mapIt = csclayermap_.begin(); mapIt != csclayermap_.end(); ++mapIt)
+	{
+	  csclayermapss<<"[CSC DetId "<<mapIt->first<<" ="<<CSCDetId(mapIt->first)<<", CSC Layer "<<mapIt->second<<" ="<<(mapIt->second)->id()<<"],"<<std::endl;
+	}
+      csclayermapss<<"]"<<std::endl;
+      std::string csclayermapstr = csclayermapss.str();
+      edm::LogVerbatim("GEMCSCSegFit") << csclayermapstr;
+      // --- End LogDebug -----------------------------------------------------------------------------------
+
+      // --- LogDebug for GEM Eta Partition map ------------------------------------------------------------
+      std::stringstream gemetapartmapss; gemetapartmapss<<"[GEMCSCSegFit::ctor] :: gemetapartmap :: elements ["<<std::endl;
+      for(std::map<uint32_t, const GEMEtaPartition*>::const_iterator mapIt = gemetapartmap_.begin(); mapIt != gemetapartmap_.end(); ++mapIt)
+	{
+	  gemetapartmapss<<"[GEM DetId "<<mapIt->first<<" ="<<GEMDetId(mapIt->first)<<", GEM EtaPart "<<mapIt->second<<"],"<<std::endl;
+	}
+      gemetapartmapss<<"]"<<std::endl;
+      std::string gemetapartmapstr = gemetapartmapss.str();
+      edm::LogVerbatim("GEMCSCSegFit") << gemetapartmapstr;
+      // --- End LogDebug -----------------------------------------------------------------------------------
+    } 
+
+  virtual ~GEMCSCSegFit() {}
+
+  // Least-squares fit
+  void fit( void ); // fill uslope_, vslope_, intercept_  @@ FKA fitSlopes()
+  // Calculate covariance matrix of fitted parameters
+  AlgebraicSymMatrix covarianceMatrix(void);
+
+  // Change scale factor of rechit x error 
+  // - expert use only!
+  void setScaleXError ( double factor ) { scaleXError_ = factor; }
+
+  // Fit values
+  float xfit( float z ) const;
+  float yfit( float z ) const;
+
+  // Deviations from fit for given input (local w.r.t. chamber)
+  float xdev( float x, float z ) const;
+  float ydev ( float y, float z ) const;
+  float Rdev( float x, float y, float z ) const;
+
+  // Other public functions are accessors
+  std::vector<const TrackingRecHit*> hits(void) const { return hits_; }
+  double scaleXError(void) const { return scaleXError_; }
+  size_t nhits(void) const { return hits_.size(); }
+  double chi2(void) const { return chi2_; }
+  int ndof(void) const { return ndof_; }
+  LocalPoint intercept() const { return intercept_;}
+  LocalVector localdir() const { return localdir_;}
+  // const CSCChamber*      cscchamber     (uint32_t id) const { return cscchambermap_.find(id)->second; }
+  const CSCChamber*      cscchamber     (uint32_t id) const { return (csclayermap_.find(id)->second)->chamber(); }
+  const CSCLayer*        csclayer       (uint32_t id) const { return csclayermap_.find(id)->second; }
+  const GEMEtaPartition* gemetapartition(uint32_t id) const { return gemetapartmap_.find(id)->second; }
+  const CSCChamber*      refcscchamber  ()            const { return (csclayermap_.find(refid_)->second)->chamber(); }
```

Comment:
The methods here on lines 130-133 need to be revised to handle error conditions. If find(id) fails to find the specified id, it returns map::end, which cannot be dereferenced. One way to fix these methods is to modify them to return a success/fail (or true/false) value along with a pointer parameter. Only if the method returns success would it be allowed to use the pointer value.